src/pyj/read_book/viewport.pyj

```diff
@@ -12,13 +12,28 @@ class ScrollViewport:
     def __init__(self):
         self.set_mode('flow')
         self.window_width_from_parent = self.window_height_from_parent = None
+        # In RTL mode, we hide the fact that we are scrolling to the left by negating the
+        # current X position and the requested X scroll position, which fools the reader
+        # code into thinking that it's always scrolling in positive X.
+        self.rtl = False
 
     def set_mode(self, mode):
         prefix = ('flow' if mode is 'flow' else 'paged') + '_'
         for attr in FUNCTIONS:
             self[attr] = self[prefix + attr]
 
+    def initialize_on_layout(self):
+        self.rtl = False
+        body_style = window.getComputedStyle(document.body)
+        if body_style.direction is "rtl":
+            self.rtl = True
```

Comment:
I hope this is relatively robust? document.body is what the handle_rtl_layout was testing for, so it seemed like a good idea to borrow that.